tests/factor/factor.py

```diff
@@ -0,0 +1,210 @@
+'''
+factor.py
+    written by Jacob Dickens, Nov 2020
+
+    Basic principle:
+        Given some integer N, find a, b, c such that
+        N = a * b * c where a, b and c are as close
+        in value as is possible.
+'''
+
+import argparse
+import typing
+
+
+def parse_args() -> argparse.Namespace:
+    '''
+    Process program arguments
+    '''
+    parser = argparse.ArgumentParser('')
+    parser.add_argument(
+        'num',
+        type=int,
+        help='Original number'
+    )
+    parser.add_argument(
+        '--dim',
+        type=int,
+        default=3,
+        help='Dimensions'
+    )
+
+    args = parser.parse_args()
+    if args.num < 1:
+        raise argparse.ArgumentTypeError(F"{args.num} isn't a positive int!")
+
+    if args.dim < 1:
+        raise argparse.ArgumentTypeError(F"{args.num} isn't a positive int")
+
+    return args
+
+
+class Factor:
```

Comment:
Why is there a lot of seemingly duplicate code here? Can you reuse the code that is in  `public/experiment.py`?